src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<K extends Machine> {
```

Comment:
```suggestion
public interface MachineProducer<T extends Machine> {
```
https://docs.oracle.com/javase/tutorial/java/generics/types.html
Pay attention to Type Parameter Naming Conventions topic

Reply:
fixed